### ISSUE TITLE
Memory overflow - PANIC

### ISSUE THREAD

**Issue by Naewis** [percona/mongodb_exporter#938]
**Describe the bug**
mongodb_exporter doesn't answer on the metric page.
Connection is cut while the exporter try to retrieve data in the mongodb server.
Logs seems to show a memory overflow.

Am I doing something wrong ? Can I do something for this issue through configuration only ?

**To Reproduce**
Basic install (See environment section to understand how it's build), but probably linked to our db size ? (obviously no-go to share the db :wink: )

**Expected behavior**
The metric page shows data :grin: 

**Logs**

Exporter error message (can't see anything else to show)
```
http: panic serving <redacted>:38996: descriptor %!s(PANIC=String method: runtime error: invalid memory address or nil pointer dereference) is invalid: (BSONObjectTooLarge) BSONObj size: 17895457 (0x1111021) is invalid. Size must be between 0 and 16793600(16MB) First element: note: "all times in microseconds"
goroutine 2153962 [running]:
net/http.(*conn).serve.func1()
	/opt/hostedtoolcache/go/1.22.8/x64/src/net/http/server.go:1903 +0xbe
panic({0xd40080?, 0xc000af34e0?})
	/opt/hostedtoolcache/go/1.22.8/x64/src/runtime/panic.go:770 +0x132
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(...)
	/home/runner/go/pkg/mod/github.com/prometheus/client_golang@v1.19.1/prometheus/registry.go:405
github.com/percona/mongodb_exporter/exporter.(*Exporter).makeRegistry(0xc0005635c0, {0x13a0498, 0xc0004feee0}, 0xc00057e0f0, {0x139cfd8, 0xc000ac2cc0}, {{0xc000484a10, 0x1, 0x1}, 0x0, ...})
	/home/runner/work/mongodb_exporter/mongodb_exporter/exporter/exporter.go:230 +0xd5f
github.com/percona/mongodb_exporter/exporter.RunWebServer.(*Exporter).Handler.func2({0x139f658, 0xc0001980e0}, 0xc00090e900)
	/home/runner/work/mongodb_exporter/mongodb_exporter/exporter/exporter.go:376 +0x6cf
net/http.HandlerFunc.ServeHTTP(0x19da000?, {0x139f658?, 0xc0001980e0?}, 0x7fc97a?)
	/opt/hostedtoolcache/go/1.22.8/x64/src/net/http/server.go:2171 +0x29
net/http.(*ServeMux).ServeHTTP(0x471879?, {0x139f658, 0xc0001980e0}, 0xc00090e900)
	/opt/hostedtoolcache/go/1.22.8/x64/src/net/http/server.go:2688 +0x1ad
net/http.serverHandler.ServeHTTP({0xc000a0b0b0?}, {0x139f658?, 0xc0001980e0?}, 0x6?)
	/opt/hostedtoolcache/go/1.22.8/x64/src/net/http/server.go:3142 +0x8e
net/http.(*conn).serve(0xc000144240, {0x13a03f0, 0xc000563950})
	/opt/hostedtoolcache/go/1.22.8/x64/src/net/http/server.go:2044 +0x5e8
created by net/http.(*Server).Serve in goroutine 12
	/opt/hostedtoolcache/go/1.22.8/x64/src/net/http/server.go:3290 +0x4b4
```

Mongo side
```
"t":{"$date":"2024-10-23T09:41:51.507+02:00"},"s":"I",  "c":"ACCESS",   "id":5286306, "ctx":"conn8802921","msg":"Successfully authenticated","attr":{"client":"<redacted>:40964","isSpeculative":true,"isClusterMember":false,"mechanism":"SCRAM-SHA-256","user":"<redacted>","db":"<redacted>","result":0,"metrics":{"conversation_duration":{"micros":7752,"summary":{"0":{"step":1,"step_total":2,"duration_micros":45},"1":{"step":2,"step_total":2,"duration_micros":23}}}},"extraInfo":{}}}
{"t":{"$date":"2024-10-23T09:41:51.507+02:00"},"s":"I",  "c":"NETWORK",  "id":6788700, "ctx":"conn8802921","msg":"Received first command on ingress connection since session start or auth handshake","attr":{"elapsedMillis":0}}
{"t":{"$date":"2024-10-23T09:41:51.509+02:00"},"s":"I",  "c":"-",        "id":20883,   "ctx":"conn8802920","msg":"Interrupted operation as its client disconnected","attr":{"opId":417696328}}
```

**Environment**
 - Mongo v7.0.9 on a `Debian GNU/Linux 12 (bookworm)` server
 - Mongo_exporter on docker 26.0.0 (image percona/mongodb_exporter:0.41.2)
 - mongo_exporter use `--mongodb.uri` with `tls=true` & `--collect-all` params, no envs vars set, mono-target

**Additional context**
I can provide more info if needed. Thx for looking into this bug.

**Comment by BupycHuk** [comment]
Please disable pbm collector for now, we are working on fix

**Comment by BupycHuk** [comment]
Fixed in v0.42.0
